### ISSUE TITLE
Use a local HTTP server for serving files in integration tests

### ISSUE THREAD

**Issue by howardjohn** [istio/istio#35835]
Some integration tests use public URLs to access files. This is problematic for a few reasons:

* Adds a dependency on external service, making tests not hermetic. This has broken us before
* Does not work in  airgap env
* Does not work for ipv6 testing.
   * In our current setup we have no external connectivity for ipv6 clusters
   * Even if we did, github does not support ipv6 it seems

Currently I know we use this for the JWKS and attributegen, but there may be others.

Proposed solution: Deploy a trivial file server that serves plaintext + HTTPS into istio-system for tests that require it

**Comment by yangminzhu** [comment]
Regarding the JWT test, we actually already have a local jwks server deployed in the cluster but it's not used for all tests, do we want to change all jwt tests to use it?

**Comment by howardjohn** [comment]
Seems reasonable.

It might be nice to also server generic bytes so we can server attributegen out of it as well (really the server is 99% there other than it serves a single hardcoded const), but we can do that in future. It might be easier to reach attributegen as well since the google API does have IPv6 connectivity at least (although that only solves the 3rd use case, but its the one I am focusing on currently).

**Comment by yangminzhu** [comment]
@aryan16 this could be a good follow-up after your last PR that adds the https support to the jwks server.

**Comment by aryan16** [comment]
Nice !! correct me if I'm wrong here, we need to add services which would minimize the external services dependencies for integration tests right ?

**Comment by nmittler** [comment]
@aryan16 @yangminzhu any update?

**Comment by nmittler** [comment]
@aryan16 is this something you'll be working on? Should we assign to you?

**Comment by aryan16** [comment]
Yeah I can look into this. I am adding one https jwks server, we can try to use it for other tests as well.

**Comment by howardjohn** [comment]
This would fix https://github.com/istio/istio/issues/38390

**Comment by aryan16** [comment]
should we use https jwks server instead of github links for pubkey ?
